node_modules/col-activities/lib/notifications/weekly.js

```diff
@@ -0,0 +1,583 @@
+ * 
+ * @param  {Number}    userId     The user id to get totals for
+ * @param  {Object}    summary    The activities summary
+ * @return {UserTotals}           Running totals for the given user
+ * @api private
+ *
+ */
+var getUserTotals = function(userId, summary) {
+  if (!_.has(summary.users, userId)) {
+    summary.users[userId] = new UserTotals();
+  }
+  return summary.users[userId];
+};
+
+/**
+ * Running totals for activites and points per asset.
+ */
+var AssetTotals = function() {
+  _.extend(this, {
+    'comments': 0,
+    'likes': 0,
+    'views': 0
+  });
+}
+
+/**
+ * Running totals for activites and points per course.
+ */
+var CourseTotals = function() {
+  _.extend(this, {
+    'pointsFromAssetsUploaded': 0,
+    'pointsFromComments': 0,
+    'pointsFromLikes': 0,
+    'pointsFromWhiteboards': 0,
+    'pointsGenerated': 0,
+    'pointsReceived': 0
+  });
+};
+
+/**
+ * Running totals for activites and points per user.
+ */
+var UserTotals = function() {
+  _.extend(this, {
+    'assets': {},
+    'commentsReceived': 0,
+    'likesReceived': 0,
+    'pointsFromAssetsUploaded': 0,
+    'pointsFromComments': 0,
+    'pointsFromLikes': 0,
+    'pointsFromWhiteboards': 0,
+    'pointsFromCommentsReceived': 0,
+    'pointsFromLikesReceived': 0,
+    'pointsCollected': 0,
+    'pointsGenerated': 0,
+    'pointsReceived': 0
+  });
+};
+
+/**
+ * Given an activity type, find the appropriate running-total property and increment by one.
+ *
+ * @param  {String}    activityType     The activity type to be translated
+ */
+ AssetTotals.prototype.incrementActivities = function(activityType) {
+  // Translate activity type to property name.
+  var property = {
+    'asset_comment': 'comments',
+    'like': 'likes',
+    'view_asset': 'views'
+  }[activityType];
+
+  // Increment the appropriate property if a translation was found.
+  if (_.has(this, property)) {
+    this[property]++;
+  }
+};
+
+/**
+ * Given an activity type, find the appropriate running-total property and increment by one.
+ *
+ * @param  {String}    activityType     The activity type to be translated
+ */
+UserTotals.prototype.incrementActivities = function(activityType) {
+  // Translate activity type to property name.
+  var property = {
+    'get_asset_comment': 'commentsReceived',
+    'get_asset_comment_reply': 'commentsReceived',    
+    'get_like': 'likesReceived'
+  }[activityType];
+
+  // Increment the appropriate property if a translation was found.  
+  if (_.has(this, property)) {
+    this[property]++;
+  }
+};
+
+/**
+ * Given an activity type and points, find the appropriate running-total property and add points.
+ * This method is used both by CourseTotals and by UserTotals.
+ *
+ * @param  {String}    activityType     The activity type to be translated
+ * @param  {Number}    points           The number of points
+ */
+CourseTotals.prototype.incrementPoints = UserTotals.prototype.incrementPoints = function(activityType, points) {
```

Comment:
@nicolaasmatthijs Perhaps just declare the function as a `var` and assign it to `that` within the `CourseTotals` and `UserTotals` constructors?